goal_src/engine/camera/cam-update.gc

```diff
@@ -5,3 +5,735 @@
+           (a2-19 (-> a3-15 vector 2 quad))
+           )
+     (set! a3-12 (-> a3-15 vector 3 quad))
+     (set! (-> v1-148 vector 0 quad) a0-54)
+     (set! (-> v1-148 vector 1 quad) a1-23)
+     (set! (-> v1-148 vector 2 quad) a2-19)
+     )
+    (set! (-> v1-148 vector 3 quad) (the-as uint128 a3-12))
+    )
+   (set! (-> *math-camera* reset) 0)
+   0
+   )
+  (let ((f0-45 (-> *math-camera* fog-min))
+        (f1-10 (-> *math-camera* fog-max))
+        )
+   (let ((f2-0 (-> *math-camera* d)))
+    (set! (-> *instance-tie-work* hmge-d x) f0-45)
+    (set! (-> *instance-tie-work* hmge-d y) f1-10)
+    (set! (-> *instance-tie-work* hmge-d z) (* 32.0 f2-0))
+    (set!
+     (-> *instance-tie-work* hmge-d w)
+     (* f2-0 (-> *math-camera* hmge-scale w))
+     )
+    (let ((v1-160 (-> *math-camera* hvdf-off quad)))
+     (set! (-> *instance-tie-work* hvdf-offset quad) v1-160)
+     )
+    (set! (-> *instance-shrub-work* hmge-d x) f0-45)
+    (set! (-> *instance-shrub-work* hmge-d y) f1-10)
+    (set! (-> *instance-shrub-work* hmge-d z) (* 3.0 f2-0))
+    (set!
+     (-> *instance-shrub-work* hmge-d w)
+     (* f2-0 (-> *math-camera* hmge-scale w))
+     )
+    (set!
+     (-> *instance-shrub-work* billboard-const x)
+     (/ (-> *math-camera* x-pix) (* f2-0 (-> *math-camera* x-ratio)))
+     )
+    (set!
+     (-> *instance-shrub-work* billboard-const y)
+     (/ (-> *math-camera* y-pix) (* f2-0 (-> *math-camera* y-ratio)))
+     )
+    )
+   (set! (-> *instance-shrub-work* billboard-const z) f0-45)
+   (set! (-> *instance-shrub-work* billboard-const w) f1-10)
+   )
+  (set!
+   (-> *instance-shrub-work* constants w)
+   (the-as float (-> *math-camera* vis-gifs 0 fog0))
+   )
+  (let ((a0-56 (-> *math-camera* hvdf-off quad)))
+   (set! (-> *instance-shrub-work* hvdf-offset quad) a0-56)
+   )
+  (update-view-planes *math-camera* (-> *math-camera* plane) 1.0)
+  (let ((t9-22 update-view-planes)
+        (a0-58 *math-camera*)
+        (a1-25 (-> *math-camera* guard-plane))
+        (a2-21 4.0)
+        )
+   (t9-22 a0-58 a1-25 a2-21)
+   (set!
+    (-> *instance-shrub-work* guard-plane 0 quad)
+    (-> *math-camera* guard-plane 0 quad)
+    )
+   (set!
+    (-> *instance-shrub-work* guard-plane 1 quad)
+    (-> *math-camera* guard-plane 1 quad)
+    )
+   (set!
+    (-> *instance-shrub-work* guard-plane 2 quad)
+    (-> *math-camera* guard-plane 2 quad)
+    )
+   (set!
+    (-> *instance-shrub-work* guard-plane 3 quad)
+    (-> *math-camera* guard-plane 3 quad)
+    )
+   (set!
+    (-> *instance-tie-work* guard-plane 0 quad)
+    (-> *math-camera* guard-plane 0 quad)
+    )
+   (set!
+    (-> *instance-tie-work* guard-plane 1 quad)
+    (-> *math-camera* guard-plane 1 quad)
+    )
+   (set!
+    (-> *instance-tie-work* guard-plane 2 quad)
+    (-> *math-camera* guard-plane 2 quad)
+    )
+   (set!
+    (-> *instance-tie-work* guard-plane 3 quad)
+    (-> *math-camera* guard-plane 3 quad)
+    )
+   (update-visible
```

Comment:
Ah, I think if you remove the extra arguments from `update-visible` the weird `a3-12` stuff will go away.